### ISSUE TITLE
Is it possible to use connectors?

### ISSUE THREAD

**Issue by syuyucheng29** [SigNoz/signoz-otel-collector#454]
I am interested in using Spanmetrics connectors to process spans, which would help me avoid exporting metrics that can be derived directly from spans.

Is it possible to enable the use of these connectors in the SigNoz collector? Or are there specific issues that may arise from enabling the connectors, which are the reason they are not intended for user access?

Any guidance or clarification on this matter would be greatly appreciated.

Thank you!

**Comment by srikanthccv** [comment]
> Is it possible to enable the use of these connectors in the SigNoz collector

Yes, they are available in signoz-otel-collector https://github.com/SigNoz/signoz-otel-collector/blob/29d5194608b3386c581175a6a3c5e76bd0e63197/components/components.go#L176-L184


> are there specific issues that may arise from enabling the connectors, which are the reason they are not intended for user access?

I am not sure what you mean by _not intended for user access_?

However, on thing to not here is we have separate version of span processor opinionated and tightly coupled with SigNoz installation because relying unstable upstream component would mean breaking things every which we wanted to avoid.

**Comment by syuyucheng29** [comment]
Thank you for responding so quickly!
 
I have tried various input configurations by adjusting the examples from the OpenTelemetry documentation but haven't had any success, which led me to mistakenly believe that connectors might not be supported in the SigNoz collector.

Thank you for clarifying that connectors are indeed available. 😊

However, during my trial-and-error process, even a simple declaration of the roundrobin connector causes the collector to stop working. For example, with the following configuration:

```conf
connectors:
  roundrobin:
```

Could you please provide a working example of how to use the Spanmetrics connector or any other connector in an otel-collector-config.yaml file? Alternatively, if that’s not possible, could you help me identify what might be wrong with the code included below?

After modifying the YAML file, I restart the collector container to apply the changes. Could this be causing any problem?

Many thanks

```conf
# After adding this code, the collector stops working.

connectors:
  spanmetrics:
    histogram:
      explicit:
        buckets: [100us, 1ms, 2ms, 6ms, 10ms, 100ms, 250ms]
    exemplars:
      enabled: true
    dimensions_cache_size: 1000
    aggregation_temporality: "AGGREGATION_TEMPORALITY_CUMULATIVE"    
    metrics_flush_interval: 15s
    metrics_expiration: 5m
    events:
      enabled: false
    resource_metrics_key_attributes:
      - service.name
      - telemetry.sdk.language
      - telemetry.sdk.name
service:
  pipelines:
    traces:
      receivers: [otlp]
      processors: [batch]
      exporters: [spanmetrics]
    metrics/test:
      receivers: [spanmetrics]
      processors: [batch]
      exporters: [clickhousemetricswrite]
```
